### PR TITLE
[Resource] Refactor ibm_pi_snapshot resource

### PR DESCRIPTION
**Pull request by ismirlia**
This PR refactors the ibm_pi_snapshot resource and cleans up the code.

```
--- PASS: TestAccIBMPIInstanceSnapshotbasic (2037.87s)
PASS
```

### CODE REVIEW

File: ibm/service/power/resource_ibm_pi_snapshot.go [hunk 1]
```diff
@@ -35,56 +35,63 @@ func ResourceIBMPISnapshot() *schema.Resource {
...
+				Description: "Description of the PVM instance snapshot.",
+				Optional:    true,
 				Type:        schema.TypeString,
+			},
+			Arg_InstanceName: {
+				Description: "The name of the instance you want to take a snapshot of.",
 				Required:    true,
-				Description: "Instance name / id of the pvm",
+				Type:        schema.TypeString,
 			},
```
[the remark this answers is not in the document]
Comment (ismirlia) [reply ibm/service/power/resource_ibm_pi_snapshot.go:54]: I think you are right; I missed it here.

**Comment by ismirlia** [comment]
```
--- PASS: TestAccIBMPIInstanceSnapshotbasic (1777.35s)
PASS
```

File: ibm/service/power/ibm_pi_constants.go [hunk 2]
```diff
@@ -7,6 +7,7 @@ const (
 	Arg_CloudConnectionName                 = "pi_cloud_connection_name"
 	Arg_CloudInstanceID                     = "pi_cloud_instance_id"
 	Arg_DatacenterZone                      = "pi_datacenter_zone"
+	Arg_Description                         = "pi_description"
```
Comment (michaelkad) [inline ibm/service/power/ibm_pi_constants.go:10]: This should be description
```suggestion
	Arg_Description                         = "description"
```

File: ibm/service/power/resource_ibm_pi_snapshot.go [hunk 3]
```diff
@@ -245,33 +251,28 @@ func isWaitForPIInstanceSnapshotAvailable(ctx context.Context, client *st.IBMPIS
...
-func isWaitForPIInstanceSnapshotDeleted(ctx context.Context, client *st.IBMPISnapshotClient, id string, timeout time.Duration) (interface{}, error) {
-
+func isWaitForPIInstanceSnapshotDeleted(ctx context.Context, client *instance.IBMPISnapshotClient, id string, timeout time.Duration) (interface{}, error) {
 	log.Printf("Waiting for (%s) to be deleted.", id)
 
-	stateConf := &resource.StateChangeConf{
-		Pending:    []string{"retry", helpers.PIInstanceDeleting},
-		Target:     []string{"Not Found"},
+	stateConf := &retry.StateChangeConf{
+		Pending:    []string{State_Retry, helpers.PIInstanceDeleting},
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_snapshot.go:274]: Could you move this the constant file? ` helpers.PIInstanceDeleting`

File: website/docs/r/pi_snapshot.html.markdown [hunk 4]
```diff
@@ -8,29 +7,32 @@ description: |-
...
 Creates, updates, deletes, and manages snapshots in the Power Virtual Server Cloud. For more information, about snapshots in the Power Virutal Server, see [snapshotting, cloning, and restoring](https://cloud.ibm.com/docs/power-iaas?topic=power-iaas-volume-snapshot-clone).
 
 ## Example usage
+
 The following example enables you to create a snapshot:
 
 ```terraform
 resource "ibm_pi_snapshot" "testacc_snapshot"{
+  pi_cloud_instance_id = "<value of the cloud_instance_id>"
+  pi_description  = "Testing snapshot for instance"
```
Comment (michaelkad) [inline website/docs/r/pi_snapshot.html.markdown:20]:
```suggestion
  description  = "Testing snapshot for instance"
```

File: website/docs/r/pi_snapshot.html.markdown [hunk 5]
```diff
@@ -47,30 +49,32 @@ The `ibm_pi_snapshot` provides the following [Timeouts](https://www.terraform.io
 - **delete** - (Default 10 minutes) Used for Deleting snapshot.
 
 ## Argument reference
+
 Review the argument references that you can specify for your resource.
- 
+
+- `pi_cloud_instance_id` - (Required, String) The GUID of the service instance associated with an account.
+- `pi_description` - (Optional, String) Description of the PVM instance snapshot.
```
Comment (michaelkad) [inline website/docs/r/pi_snapshot.html.markdown:56]:
```suggestion
- `description` - (Optional, String) Description of the PVM instance snapshot.
```

**Review by michaelkad — CHANGES_REQUESTED** [review]
Changing to pi_description would be a breaking change. Keeping it as description, we can add ` pi_description` along side `description` but as a deprecated field instead if you wish to go that route

File: ibm/service/power/resource_ibm_pi_snapshot.go (hunk 3 above)
Comment (ismirlia) [reply ibm/service/power/resource_ibm_pi_snapshot.go]: Good catch

File: website/docs/r/pi_snapshot.html.markdown [hunk 6]
```diff
@@ -47,30 +49,32 @@ The `ibm_pi_snapshot` provides the following [Timeouts](https://www.terraform.io
...
 
-The `ibm_pi_snapshot` resource can be imported by using `power_instance_id` and `snapshot_id`.
+The `ibm_pi_snapshot` resource can be imported by using `cloud_instance_id` and `snapshot_id`.
 
-**Example**
+### Example
 
-```
-$ terraform import ibm_pi_snapshot.example d7bec597-4726-451f-8a63-e62e6f19c32c/cea6651a-bc0a-4438-9f8a-a0770bbf3ebb
+```base
```
Comment (michaelkad) [inline website/docs/r/pi_snapshot.html.markdown:78]:
```suggestion
```bash
```

**Review by michaelkad — CHANGES_REQUESTED** [review]
A typo -- all good other than that

**Review by michaelkad — APPROVED** [review]

**Comment by ismirlia** [comment]
Opened in main terraform repo: https://github.com/IBM-Cloud/terraform-provider-ibm/pull/5331

**Pull request closed without merging**